crates/redstone/src/core/config.rs

```diff
@@ -30,3 +35,242 @@ pub struct Config {
     /// for determining whether the data is current in the context of blockchain time.
     pub block_timestamp: TimestampMillis,
 }
+
+impl Config {
+    /// Verifies all members of the config.
+    ///
+    /// This method checks whether all configs members are correct.
+    ///
+    /// # Returns
+    ///
+    /// * Success `()` if config is valid or Err with `Error` otherwise.
+    pub(crate) fn verify(&self) -> Result<(), Error> {
```

Comment:
This comment will be about somethign larger, and that we can do it in the followups - since it will break api

We should disallow creation of the invalid configs, what it means:
* make function `Config::new() -> Result<Config>`.
* make it the only way to create config
* get rid of public fields in the config.

Reply:
Good idea. What do you think to name it `Config::try_new() -> Result<Config>`

Reply:
sure, better name :)